tutorials/persistance-comparison-series/File/file.md

```diff
@@ -0,0 +1,345 @@
+
+```cs
+private class HitCount
+{
+    public int Unmodified;
+    public int Shift;
+    public int Control;
+}
+```
+
+The class itself can be `private` and just be added inside the `FileExampleJson` class but its fields need to be public.
+
+As before we use a different file to save this data. Update the filename to:
+
+```cs
+private const string HitCountFileJson = "hitCountFileJson.txt";
+```
+
+When saving the data we will use the same `Update()` method as before to detect which key was pressed.
+
+The first part of `OnMouseDown()` (1) can stay the same as well since this part only increments the hit count in depending on the modifier used.
+
+```cs
+private void OnMouseDown()
+{
+    // 1
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift)
+    {
+        // Increment the Shift hit count.
+        hitCountShift++;
+    }
+    else if (keyPressed == KeyCode.LeftControl)
+    {
+        // Increment the Control hit count.
+        hitCountControl++;
+    }
+    else
+    {
+        // If neither Shift nor Control was held, we increment the unmodified hit count.
+        hitCountUnmodified++;
+    }
+
+    // 2
+    // Create a new HitCount object to hold this data.
+    HitCount hitCount = new();
+    hitCount.Unmodified = hitCountUnmodified;
+    hitCount.Shift = hitCountShift;
+    hitCount.Control = hitCountControl;
+
+    // 3
+    // Create a JSON using the HitCount object.
+    string jsonString = JsonUtility.ToJson(hitCount, true);
+
+    // 4
+    // Save the json to the file.
+    File.WriteAllText(HitCountFileJson, jsonString);
+}
+```
+
+However, we need to update the second part. Instead of a string array we create a new `HitCount` object and set the three public fields to the values of the hit counters (2).
+
+Using `JsonUtility.ToJson()` we can transform this object to a string (3). If you pass in `true` for the second, optional parameter `prettyPrint` the string will be formatted in a nicely readable way.
+
+Finally, as in `FileExampleSimple.cs` we just use `WriteAllText()` since we're only saving one string, not an array.
+
+When you run the game you will the that in the editor it looks identical to the previous section since we are using the same three counters. If you open the file `hitCountFileJson.txt` you should then see the three counters in a nicely formatted JSON.
+
+<img src="images/09_hit_count_extended_json.jpg" alt="File hitCountExtended.txt" width=""/>
+
+## Conclusion
+
+In this tutorial we learned how to utilize `File` to save data. `JsonUtility` helps structure this data. They are simple and easy to use, not much code is required.
+
+What are the down sides though?
+
+First of, we open, write to and save the file every single time the capsule it clicked. While not a problem in this case and certainly applicable for some games this will not perform very well when many save operations are made.
```

Comment:
```suggestion
First of all, we open, write to, and save the file every single time the capsule it clicked. While not a problem in this case and certainly applicable for some games this will not perform very well when many save operations are made.
```